src/@context/Asset.tsx

```diff
@@ -78,7 +80,32 @@ function AssetProvider({
             '\n\nWe could not find an asset for this DID in the cache. If you just published a new asset, wait some seconds and refresh this page.'
         )
         LoggerInstance.error(`[asset] Failed getting asset for ${did}`, asset)
-      } else {
+        return
+      }
+
+      if (asset.nft.state) {
+        // handle nft states as documented in https://docs.oceanprotocol.com/concepts/did-ddo/#state
+        let state
+        switch (asset.nft.state) {
+          case 1:
+            state = 'end-of-life'
+            break
+          case 2:
+            state = 'deprecated'
+            break
+          case 3:
+            state = 'revoked'
+            break
+        }
+
+        setTitle(`This asset has been flagged as "${state}" by the publisher`)
+        setIsV3Asset(await checkV3Asset(did, token))
+        setWarning(`\`${did}\`` + `\n\nPublisher Address: ${asset.nft.owner}`)
+        LoggerInstance.error(`[asset] Failed getting asset for ${did}`, asset)
+        return
```

Comment:
Just double checking, are `end-of-life` assets indexed or not? Cause in that casde returning here will not show the asset. Do we even have docs on what exactly the states should imply?

Reply:
not sure what implies, I couldn't find anything useful on the docs, but maybe I'm just not looking in the correct place. @mihaisc do you know where we can find this info?

Reply:
to quote myself `1,2,3 - same action just different message` . The metadata is not indexed.